pypsa/definitions/components.py

```diff
@@ -1,120 +1,126 @@
+        if not isinstance(other, ComponentTypeInfo):
+            return NotImplemented
 
-    def __repr__(self) -> str:
         return (
-            f"Component(name={self.name!r}, list_name={self.list_name!r}, "
-            f"attrs=Keys({list(self.attrs.keys())}), static=DataFrame(shape={self.static.shape}), "
-            f"dynamic=Keys({list(self.dynamic.keys())}))"
+            self.name == other.name
+            and self.list_name == other.list_name
+            and self.description == other.description
+            and str(self.category) == str(other.category)
+            and self.defaults.equals(other.defaults)
         )
 
-    # @deprecated(
-    #     deprecated_in="0.32",
-    #     removed_in="1.0",
-    #     details="Use `c.static` instead.",
-    # )
+    def __repr__(self) -> str:
+        return self.name + " Component Type"
+
     @property
-    def df(self) -> pd.DataFrame:
-        return self.static
-
-    # @deprecated(
-    #     deprecated_in="0.32",
-    #     removed_in="1.0",
-    #     details="Use `c.dynamic` instead.",
-    # )
+    @deprecated(
+        deprecated_in="0.32.0",
+        details="Use the 'category' attribute instead.",
+    )
+    def type(self) -> str:
+        return self.category
+
     @property
-    def pnl(self) -> Dict:
-        return self.dynamic
+    @deprecated(
+        deprecated_in="0.32.0",
+        details="Use the 'defaults' attribute instead.",
+    )
+    def attrs(self) -> pd.DataFrame:
+        return self.defaults
+
+
+class ComponentsStore(dict):
+    def __setattr__(self, name: str, value: Any) -> None:
+        if hasattr(ComponentsStore, name):
+            msg = "'ComponentsStore' object attribute " f"'{name}' can not be set."
+            raise AttributeError(msg)
+        self[name] = value
+
+    def __getattr__(self, item: str) -> Any:
+        """
+        Get attribute from the dictionary.
+
+        Examples
+        --------
+        >>> components = ComponentsStore()
+        >>> components["generator"] = Generators()
+        >>> components.generators
+        """
+        try:
+            return self[item]
+        except KeyError:
+            msg = f"'{self.__class__.__name__}' object has no component '{item}'."
+            raise AttributeError(msg)
+
+    def __getitem__(self, item: str | Sequence) -> Any:
+        """
+        Index single and multiple items from the dictionary.
+
+        Similar behavior to pandas.DataFrame.__getitem__.
+
+        Examples
+        --------
+        >>> components = ComponentsStore()
+        >>> components["generator"] = Generators()
+        >>> components["storage"] = Storage()
+        >>> components["generator"]
+        'Generator class instance'
+        >>> components[["generator", "storage"]]
+        >>> components[["generator", "storage"]]
+        ['Generator class instance', 'Storage class instance']
+        """
+        if isinstance(item, (list, set)):
+            return [self[key] for key in item]
```

Comment:
Hmm. I see, but not sure if I agree. For DataFrame consistency it makes sense and for them it also is useful since you wanna use DataFrame specific features on your sliced `df`. 
But here there is no use case where you would need any ComponentsStore functionality on a sliced version. And then it just complicates things for no reason. If this returns a list, the ComponentsStore is always a complete representation of all components for a specific Network.